Expensify/App
# [No QA] #4819 improve the check port script

### Details

I personally had a number of issues with the original implementation of the script. This PR fixes them by rewriting that script with built-in `isPackagerRunning` helper.

### Fixed Issues

$ https://github.com/Expensify/App/issues/4819

### Tests


**You can use `npm run ios` or `npm run android` in the examples below**

Test 1 (if you are not affected by an issue like this https://github.com/facebook/react-native/issues/28807 and metro bundler opens normally on other react native projects in your environment):
1. Manually ensure no servers are running on port 8081.
2. Run `npm run ios`.
3. The metro bundler should start normally .

Test 2:
1. Start metro bundler manually: `npm start`.
2. Run `npm run ios`.
3. There should be no errors, no new bundler should spawn and the app should run properly.

Test 3:
1. Start webpack for web and desktop: `npm run desktop` and `npm run web`.
2. Run `npm run ios`.
3. You should see an error listing. Somewhere inside it you will find the following error message: 
```
The port 8081 is currently in use. You can run `lsof -i :8081` to see which program is using it.
```

### Tested On

- [x] GNU/Linux
